extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -130,6 +154,13 @@ function SidebarWithRef(
       return;
     }
 
+    const document = sidebarElement.ownerDocument;
```

Comment:
So, yeah, I think this warrants two layout effects:

The first one:

```
useLayoutEffect(() => {
  if (side) {
    // Already computed. Bail out.
    return;
  }
  ... calculateSide(...) ...
}, [side]);
```

And second for animation:
```
useLayoutEffect(() => {
  if (!side) {
    // Not yet ready to for animation. Wait until this is updated by the first layout effect.
    // TODO: may also need to set visibility hidden here.
    return;
  }
}, [...., side]);
```

Reply:
Sounds good, latest revision handles setting `side` state in one `layoutEffect` and animations in a separate one.  Comments directly in code for the new revision!  Thanks!